src/bidiMapper/modules/input/ActionDispatcher.ts

```diff
@@ -90,21 +91,33 @@ export class ActionDispatcher {
     return result.result.value;
   };
 
+  readonly #browsingContextStorage: BrowsingContextStorage;
+
   #tickStart = 0;
   #tickDuration = 0;
   #inputState: InputState;
-  #context: BrowsingContextImpl;
+  #contextId: string;
   #isMacOS: boolean;
+
   constructor(
     inputState: InputState,
-    context: BrowsingContextImpl,
+    browsingContextStorage: BrowsingContextStorage,
+    contextId: string,
     isMacOS: boolean,
   ) {
+    this.#browsingContextStorage = browsingContextStorage;
```

Comment:
We could pass a delegate for getting the context, or add the logic to the browsing context to throw exception when accessing it's target when the browsing context is disposed. But I think this way is the most readable.